docs/build-results-archive.md

```diff
@@ -0,0 +1,52 @@
+# Build results archive.
+
+This document describes a build results archive format which can be used to store the sarif results of a build-and-test procedure.
+
+The archive will take the form of a compressed (zstd if available, or bzip2 otherwise) tar file containing the following files.
+
+```
+./build-results-archive
+./build/**/*.sarif
+./colcon-build-cmd
+./colcon-test-cmd
+./processed.sarif
+./vcs-export-exact.repos
+```
+
+The individual files are further described:
+
+## build-results-archive
+
+This is a JSON file containing metadata regarding the current archive.
+The file _must_ contain an single top-level JSON object and that object _must_ contain the `"version"` key with the value of `1`.
+Future revisions of this format may add additional valid versions.
+
+## build/**/*.sarif
+
+The build/ subdirectory _must_ contain the unaltered sarif content of the workspace build directory after the end of the latest colcon test run.
+
+## colcon-build-cmd
+
+This file _must_ contain the full colcon build command described in the colcon logs under the `latest_build` symlink.
+The file, when run as a /bin/sh program or prefixed with the `exec` shell command in a workspace root should invoke colcon in the local environment using the same arguments as the original build.
+
+## colcon-test-cmd
+
+This file is identical in behavior to `colcon-build-cmd` but should describe the test command under the `latest_test` symlink.
+
+## processed-sarif-cmd
+
+This file is optional but recommended.
+If it is present, the processed.sarif file _must_ also be present.
+If present it should be identical to the colcon-build-cmd file in format but describe instead the command used to generate the processed.sarif file.
+
+## processed.sarif
```

Comment:
Is this something we want to have? Merging Sarif files is non-trivial, but we could implement it.

Pros:
- Single file to open for the Dashboard to process and produce a view for
- Single place for developers to look for output if they want a manual view

Cons:
- A bit tricky to implement
    - What to do with multiple tools?
    - Need to merge Artifact lists and update Result indices accordingly
    - Need to ensure SARIF metadata is set properly (high level `properties` field)

It may be as simple as appending multiple objects to the `runs` list, but currently `process_sarif` assumes a single run per SARIF file.

Reply:
+1 for not merging the files

Reply:
Okay it sounds like from out-loud discussion that keeping the same file structure (one output per tool per package) in the processed sarif is the way to go. This means that rather than have one processed.sarif file we'll have a directory mirroring the build directory but with de-duplicated sarif files.

Reply:
https://github.com/space-ros/process_sarif/pull/6/commits/2d3b4fa30c36f6680f170652e8e1f198868ed6aa updates the doc to specify a processed directory rather than a single file.